cartographer_rviz/cartographer_rviz/drawable_submap.cc

```diff
@@ -159,13 +168,16 @@ void DrawableSubmap::SetAlpha(const double current_tracking_z) {
 
 void DrawableSubmap::UpdateSceneNode() {
   ::cartographer::common::MutexLocker locker(&mutex_);
+  render_in_progress_ = false;
+  if (quitting_) {
```

Comment:
Isn't the only benefit of adding the logic for `quitting_` and `render_in_progress_` to save a tiny bit of performance? I'd then argue for keeping the current code. That it finishes the query by updating the texture which is subsequently destroyed seems a tiny bit more work in exchange for simpler code.